docs/REST API/rest-api.md

```diff
@@ -6936,12 +6938,74 @@ Deletes the session specified by ID. When the user associated with the session n
 
 ## Software
 
+- [Upload a software](#upload-a-software)
+- [Delete a software](#delete-a-software)
 - [List software titles](#list-software-titles)
 - [List software versions](#list-software-versions)
 - [Count software](#count-software)
 - [Get software title](#get-software-title)
 - [Get software version](#get-software-version)
 
+### Upload a software
+
+Upload a software to install on macOS and Windows hosts during enrollment.
+
+`POST /api/v1/fleet/software`
```

Comment:
```suggestion
`POST /api/v1/fleet/software/upload`
```

I wonder if we should have a more explicit name here and add `/upload`. I like the simple `POST /software` but I wonder if we should be more explicit for the API user. 

This is the endpoint you use to upload software to Fleet. Same thing for `DELETE /software/upload`. As an API user, I'm not deleting the software entry from Fleet. 

If it's installed on hosts it will will still show up in Fleet. I'm deleting the software upload. 

Let's loop in @rachaelshaw on this one too.

What do y'all think?

Reply:
Good point. I think we should differentiate this from regular software that's just read-only.